src/sentry/integrations/slack/tasks.py

```diff
@@ -170,6 +176,20 @@ def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=No
             redis_rule_status.set_value("failed")
             return
 
+    # This function call will write "input_channel_id" to the appropriate actions before we create the serializer
+    # I wasn't sure of a way to use the same loop to return True in the endpoints and to also do the lookup and didn't want to duplicate the loop
+    try:
+        mapped_ids = get_slack_channel_ids(organization, user, data)
+    except (serializers.ValidationError, ChannelLookupTimeoutError, InvalidTriggerActionError):
+        # channel doesn't exist error or validation error
+        redis_rule_status.set_value("failed")
+        return
+
+    for trigger in data["triggers"]:
+        for action in trigger["actions"]:
+            if action["targetIdentifier"] in mapped_ids:
```

Comment:
@iProgramStuff what happens if the target identifier isn't in our `mapped_ids`? Doesn't that mean we should return a failure?